src/settings/s_decisions.php

```diff
@@ -147,7 +147,80 @@ function render(SettingValues $sv) {
         "<p class=\"settingtext\">To collect <em>multiple</em> final versions, such as one in 9pt and one in 11pt, add “Alternate final version” options via <a href='", hoturl("settings", "group=opt"), "'>Settings &gt; Submission options</a>.</p>",
         "</div>\n\n";
     Ht::stash_script("foldup(\$\$('cbfinal_open'),null,{f:\"c\"})");
+
+	//Share with Open Academic
+    echo "<h3 class=\"settings g\">Share with search engines</h3>\n";
+    echo "<p><b>Click the button to share accepted paper meta data with search engines</b></p>";
+    echo Ht::submit("share", "Share with search engines");
+
+    //wakes up API
+    $api = 'https://openacademicapi.azurewebsites.net/api/papers';
+    $opts = array('http'=> array('method' => 'GET'));
+    $context = stream_context_create($opts);
+    $post_result = file_get_contents($api, false, $context);
+   
+    if (isset($_REQUEST["share"])) {
+        $this->share($sv, $api);
+    }
+
 }
+    //sends https post request to the api
+    function share(SettingValues $sv, $api) {
+
+        $share = array();
+        $share["provider"] = "HotCRP";
+        $share["setId"] = "";
+        $share["venue"] = $sv->conf-> long_name;
+        $share["shortVenue"] = $sv->conf-> short_name;
+        $share["confSite"] = $sv->conf-> opt["conferenceSite"];
+        $share["version"] = "1.0";
+    
+        $papers = array();
+        $result = $sv->conf-> q_raw("SELECT paperID, title, authorInformation, abstract FROM Paper WHERE outcome > 0");
+        while (($row = edb_row($result))) {
+            $curr_p = [];
+            $curr_p['paperId'] = $row[0];
+            $curr_p['title'] = $row[1];
+            $curr_p['abstract'] = $row[3];
+            
+            //parse the authorInformation into an array
+            $authors = [];
+            $mul_authorRow = explode("\n", $row[2]);
+            
+            //the last element in the array is null, do not want the null
+            $mul_authorRow = array_slice($mul_authorRow, 0, count($mul_authorRow) - 1);
+            
+            foreach ($mul_authorRow as $sing_authorRow) {
+                $authorInfo = explode("\t", $sing_authorRow);
+                $curr_author = [];
+                //combine first name and last name to full name
+                $curr_author['name'] = $authorInfo[0]." ". $authorInfo[1];
+                if (strlen($authorInfo[2]) == 0) {
+                   $curr_author['srcId'] = $authorInfo[2];
+                } else {
+                   $curr_author['srcId'] = sha1($authorInfo[2]);
```

Comment:
Hm. I’m a little confused by this for a couple reasons.

What is the intention here? To uniqueify different authors?

1. Remember that email addresses are case-insensitive. If you really want uniqueness, so the same email address has the same `srcId` across different conferences, do `strtolower` on the email.
2. The sha1 will make it hard to extract an email address from the API's results, but will make it easy to verify an email address's presence in the API (since anyone can sha1 an email). So not clear what your goal is here.
3. People often give variant email addresses (`kaashoek@mit.edu`, `kaashoek@csail.mit.edu`, `kaashoek@pdos.csail.mit.edu`). They'll be considered different.
4. The name `srcId` for this field is nasty & unclear.